apps/src/javalab/JavabuilderConnection.js

```diff
@@ -489,4 +517,24 @@ export default class JavabuilderConnection {
       }
     );
   }
+
+  resetRunState() {
+    this.seenMessage = false;
+    this.hadWebsocketConnectionError = false;
+    this.sawValidationTests = false;
+    this.allValidationPassed = true;
+    this.seenUnsupportedNeighborhoodMessage = false;
+    this.seenUnsupportedTheaterMessage = false;
+  }
+
+  toggleOffRunningOrTesting() {
```

Comment:
nit maybe outside of scope: I think of "toggling" as necessarily switching a binary state. Since this looks like it can only turn things off, should this be called `setOffRunningOrTesting` or `turnOffRunningOrTesting`?

Reply:
will update!